dpnp/dpnp_iface_arraycreation.py

```diff
@@ -501,6 +697,21 @@ def asfortranarray(
     >>> x is y
     True
 
+    Creating an array on different device and usm_type
+
+    >>> x0 = np.asarray([1, 2, 3])
+    >>> x = np.asfortranarray(x0)
+    >>> x, x.device, x.usm_type
+    (array([1, 2, 3]), Device(level_zero:gpu:0), 'device')
+
+    >>> y = np.asfortranarray(x0, device="cpu")
+    >>> y, y.device, y.usm_type
+    (array([1, 2, 3]), Device(opencl:cpu:0), 'device')
+
+    >>> z = np.asfortranarray(x0, usm_type="host")
+    >>> z, z.device, z.usm_type
+    (array([1, 2, 3]), Device(level_zero:gpu:0), 'host')
+
     """
 
```

Comment:
This function works not only with these types. I updated the description of the parameters

Reply:
Since we check `a.ndim = 0`, before calling `asarray`, it will fail for lists, tuples, etc as input. you may want to modify it to call `a.ndim` only when input is an array. The same modification should be implemented in `ascontiguousarray`.
And if it is needed add proper test to have this change covered.

Reply:
Fix in #1691